flink/src/main/java/org/apache/iceberg/flink/data/vectorized/VectorizedFlinkOrcReaders.java

```diff
@@ -0,0 +1,278 @@
+          fieldVectors[pos] = fieldConverters.get(pos)
+              .convert(structVector.fields[pos], batchSize, batchOffsetInFile);
+        }
+      }
+
+      return new FlinkRowColumnVector(fieldVectors, structVector);
+    }
+
+    private ColumnVector toConstantColumnVector(Type type, Object constant) {
+      Type.TypeID typeID = type.typeId();
+      switch (typeID) {
+        case INTEGER:
+        case DATE:
+        case TIME:
+          return ConstantColumnVectors.ints(constant);
+
+        case LONG:
+          return ConstantColumnVectors.longs(constant);
+
+        case BOOLEAN:
+          return ConstantColumnVectors.booleans(constant);
+
+        case DOUBLE:
+          return ConstantColumnVectors.doubles(constant);
+
+        case FLOAT:
+          return ConstantColumnVectors.floats(constant);
+
+        case DECIMAL:
+          return ConstantColumnVectors.decimals(constant);
+
+        case TIMESTAMP:
+          return ConstantColumnVectors.timestamps(constant);
+
+        case FIXED:
+        case UUID:
+        case BINARY:
+        case STRING:
+          return ConstantColumnVectors.bytes(constant);
+
+        default:
+          throw new UnsupportedOperationException("Unsupported data type for constant.");
+      }
+    }
+
+    private static class ArrayConverter implements Converter {
+      private final Converter elementConverter;
+
+      private ArrayConverter(Converter elementConverter) {
+        this.elementConverter = elementConverter;
+      }
+
+      @Override
+      public ColumnVector convert(org.apache.orc.storage.ql.exec.vector.ColumnVector vector, int batchSize,
+                                  long batchOffsetInFile) {
+        ListColumnVector listVector = (ListColumnVector) vector;
+        ColumnVector elementVector = elementConverter.convert(listVector.child, batchSize, batchOffsetInFile);
+
+        return new ArrayColumnVector() {
+          @Override
+          public ArrayData getArray(int rowId) {
+            int index = getRowIndex(rowId);
+            return new ColumnarArrayData(elementVector, (int) listVector.offsets[index],
+                (int) listVector.lengths[index]);
+          }
+
+          @Override
+          public boolean isNullAt(int rowId) {
+            return vector.isNull[getRowIndex(rowId)];
+          }
+
+          private int getRowIndex(int rowId) {
+            return vector.isRepeating ? 0 : rowId;
+          }
+        };
+      }
+    }
+  }
+
+  private static class FlinkRowColumnVector implements RowColumnVector {
+
+    private final ColumnVector[] fieldVectors;
+    private final StructColumnVector structVector;
+    private final VectorizedColumnBatch vectorizedColumnBatch;
+
+    FlinkRowColumnVector(ColumnVector[] fieldVectors,
+                         StructColumnVector structVector) {
+      this.fieldVectors = fieldVectors;
+      this.structVector = structVector;
+      vectorizedColumnBatch = new VectorizedColumnBatch(fieldVectors);
```

Comment:
Nit:  It will be good if we could algin the code style with the above line :  `this.vectorizedColumnBatch=new VectorizedColumnBatch(fieldVectors)`.